--- mod_perl-1.19/src/modules/perl/Connection.xs.ipv6	Fri Jan  8 17:52:40 1999
+++ mod_perl-1.19/src/modules/perl/Connection.xs	Tue Jul 13 12:52:18 1999
@@ -62,7 +62,7 @@
     RETVAL = newSVpv((char *)&conn->remote_addr,
                       sizeof conn->remote_addr);
     if(sv_addr) {
-        struct sockaddr_in addr; 
+        struct sockaddr_storage addr; 
         STRLEN sockaddrlen; 
         char * new_addr = SvPV(sv_addr,sockaddrlen); 
         if (sockaddrlen != sizeof(addr)) { 
