--- mod_perl-1.99_17/lib/ModPerl/BuildMM.pm~	2004-09-14 21:30:41.000000000 +0200
+++ mod_perl-1.99_17/lib/ModPerl/BuildMM.pm	2004-11-28 21:51:19.000000000 +0100
@@ -191,7 +191,7 @@
     # add the code to glue the existing pods to the .pm files in blib.
     # create a dependency on pm_to_blib subdirs linkext targets to
     # allow 'make -j'
-    my @target = ('glue_pods: pm_to_blib subdirs linkext');
+    my @target = ('glue_pods: pm_to_blib.ts subdirs linkext');
 
     if (-d $doc_root) {
         my $build = build_config();
