diff -urN mod_perl-1.99_17_org/lib/Apache/Build.pm mod_perl-1.99_17/lib/Apache/Build.pm
--- mod_perl-1.99_17_org/lib/Apache/Build.pm	2004-09-26 09:06:23.000000000 +0200
+++ mod_perl-1.99_17/lib/Apache/Build.pm	2004-11-09 18:41:20.043174376 +0100
@@ -1499,16 +1499,16 @@
     if (!$self->should_build_apache) {
         $install .= <<'EOI';
 # install mod_perl.so
-	@$(MKPATH) $(MODPERL_AP_LIBEXECDIR)
+	@$(MKPATH) $(DESTDIR)$(MODPERL_AP_LIBEXECDIR)
 	$(MODPERL_TEST_F) $(MODPERL_LIB_DSO) && \
-	$(MODPERL_CP) $(MODPERL_LIB_DSO) $(MODPERL_AP_LIBEXECDIR)
+	$(MODPERL_CP) $(MODPERL_LIB_DSO) $(DESTDIR)$(MODPERL_AP_LIBEXECDIR)
 EOI
     }
     
     $install .= <<'EOI';
 # install mod_perl .h files
-	@$(MKPATH) $(MODPERL_AP_INCLUDEDIR)
-	$(MODPERL_CP) $(MODPERL_H_FILES) $(MODPERL_AP_INCLUDEDIR)
+	@$(MKPATH) $(DESTDIR)$(MODPERL_AP_INCLUDEDIR)
+	$(MODPERL_CP) $(MODPERL_H_FILES) $(DESTDIR)$(MODPERL_AP_INCLUDEDIR)
 EOI
 
     my $mf = $self->default_file('makefile');
diff -urN mod_perl-1.99_17_org/Makefile.PL mod_perl-1.99_17/Makefile.PL
--- mod_perl-1.99_17_org/Makefile.PL	2004-09-26 09:10:55.000000000 +0200
+++ mod_perl-1.99_17/Makefile.PL	2004-11-09 18:07:29.142918144 +0100
@@ -70,7 +70,7 @@
     VERSION       => $VERSION,
     NO_META       => 1,
     ABSTRACT_FROM => 'lib/mod_perl.pm',
-    EXE_FILES     => \@exe_files,
+#    EXE_FILES     => \@exe_files,
     DEFINE        => get_DEFINE(),
     macro         => {
         MODPERL_SRC        => $code->path,
@@ -150,7 +150,7 @@
                 error "cannot install mod_perl/$VERSION on top of $vstring",
                     "use MP_INST_APACHE2=1 option " .
                      "or to force installation delete:\n $old_modperl_pm";
-                exit 1;
+#                exit 1;
             }
         }
     }
@@ -591,8 +591,8 @@
 	cd "$(MODPERL_SRC)" && $(MAKE) install
 
 modperl_xs_h_install:
-	@$(MKPATH) $(MODPERL_AP_INCLUDEDIR)
-	$(CP) $(MODPERL_XS_H_FILES) $(MODPERL_AP_INCLUDEDIR)
+	@$(MKPATH) $(DESTDIR)$(MODPERL_AP_INCLUDEDIR)
+	$(CP) $(MODPERL_XS_H_FILES) $(DESTDIR)$(MODPERL_AP_INCLUDEDIR)
 
 modperl_src_clean:
 	cd "$(MODPERL_SRC)" && $(MAKE) clean
